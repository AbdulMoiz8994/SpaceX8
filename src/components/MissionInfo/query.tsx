import gql from 'graphql-tag'


export const QUERY_LAUNCHDERAILS=gql`
   query launchID($id: String!) {
  launch(id: $id) {
    launch_success
    launch_year
    mission_name
    details
    rocket {
      rocket_name
      rocket_type
    }
    launch_site {
      site_name
    }
    links {
      flickr_images
      video_link
      wikipedia
    }
  }
}`