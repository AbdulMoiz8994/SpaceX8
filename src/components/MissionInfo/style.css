@import url('https://fonts.googleapis.com/css2?family=Gemunu+Libre:wght@300&display=swap');



.mainMissionInfo{
    margin-left: 60px;
    border: 2px solid gray;
    width: 50vw;

}
.MissionnesteDiv{

}

