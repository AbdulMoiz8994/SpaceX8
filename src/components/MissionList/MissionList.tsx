import React from 'react'
import {MissionListQuery} from '../../generated/graphql'
import './styles.css'
import spacex from '../Image/logo192.png'

interface Props{
    data: MissionListQuery | undefined
}


export const MissionList: React.FC<Props> = ({data}) => {
    console.log(data);
    
    return (
        <div className="mainDiv">
          <div className="first_child">
              {/* <h1>SpaceX Missions</h1>
               */}
               <img src={spacex} alt="" />
              <div className="listDiv">
            <ul className="UL">
            <div className="list_items">
                  {data?.launches?.map((value, id) =>{
                    return( <li key={id}><a href="#/">{value?.mission_name}</a></li>)
                  })}
                  </div>
            </ul>
            </div>
              </div>           
        </div>
    )
}
