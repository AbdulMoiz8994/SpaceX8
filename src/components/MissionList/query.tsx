import gql from 'graphql-tag';

export const  MissionQuery=gql`
  query MissionList{
    launches {
    mission_id
    mission_name
    launch_success
    launch_year 
  }
  }`


