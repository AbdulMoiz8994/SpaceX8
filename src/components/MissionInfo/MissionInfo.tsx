import React from 'react'
import {LaunchIdQuery} from '../../generated/graphql'
import './style.css'

interface Prop{
    data: LaunchIdQuery | undefined
}


export const MissionInfo:React.FC<Prop> = ({data}) => {
console.log(data);

// loading Sate for Server is coming
if(!data?.launch){
    return (<h1>The Data is Unavailable</h1>)
}


    return (
        <div className="mainMissionInfo">
            <div className="MissionnesteDiv">
                <h1>Mission Details</h1>
                <span>Mission: {data.launch.mission_name}</span>
            </div>
        </div>
    )
}
