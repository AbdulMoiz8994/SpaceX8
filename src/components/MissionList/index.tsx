import React from 'react'
import {useMissionListQuery} from '../../generated/graphql'
import {MissionList} from './MissionList'
import  {OwnProps} from './MissionList'

  

export const MissionContainer = (props: OwnProps) => {
    const {data,loading,error}=useMissionListQuery()

    if(loading){
        return <h1>The Data is Loading</h1>
    }
    if(error){
        return <h1>There is an Error</h1>
    }
    //  console.log(data);
     
    return (
        <div>
            <MissionList  data={data} {...props} />
        </div>
    )
}
