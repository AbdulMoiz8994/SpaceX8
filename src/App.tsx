import React from 'react';
import './App.css';
import {MissionContainer} from './components/MissionList/index'
import {MissionInfoCont} from './components/MissionInfo/index'
function App() {
  return (
    <div className="App">
        <MissionContainer/>
        <div>
        <MissionInfoCont/>
        </div>
        
    </div>
  );
}

export default App;
