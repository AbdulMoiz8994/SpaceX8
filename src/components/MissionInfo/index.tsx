import React, { useEffect } from 'react';
import {useLaunchIdQuery} from '../../generated/graphql'
import {MissionInfo} from './MissionInfo'


interface Props{
    id: number
}


export const MissionInfoCont=({id}:Props) =>{


   const {data,loading,error}=useLaunchIdQuery({
       variables:{
        //    we add String because the id key has string type and we need to convert it number to string
           id: String(id)
       }
   });
   if(loading){
       return <h1>The Data is Loading</h1>
   }
   if(error){
       return <h1>There is an Error</h1>
   }
   console.log(data);
   

   return(
       <div>
          <MissionInfo data={data}/>
       </div>
   )
}