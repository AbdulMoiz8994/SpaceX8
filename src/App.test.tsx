import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import {MissionInfoCont} from './components/MissionInfo/index'
import {MissionContainer} from './components/MissionList/index'

describe('App', () => {
  const container = shallow(<App />);
	it('should render a <div/>', () => {
		expect(container.find('div').length).toEqual(3);
});
   it('should Reder the First Comp' , () =>{
     expect(container.containsMatchingElement(<MissionContainer handleChange={jest.fn()}/>))
   })
 
  it('Checking the  MissionInfo Comp',() =>{
    expect(container.containsMatchingElement(<MissionInfoCont id={100}/>)).toEqual(true)

  })
   it('This time is making the snapshot',() =>{
     expect(container).toMatchSnapshot()
   })
});