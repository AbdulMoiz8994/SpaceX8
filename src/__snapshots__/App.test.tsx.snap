// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App This time is making the snapshot 1`] = `
<div
  className="App"
>
  <div>
    <MissionContainer
      handleChange={[Function]}
    />
  </div>
  <div>
    <MissionInfoCont
      id={100}
    />
  </div>
</div>
`;
